flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImpl.java

```diff
@@ -773,15 +788,9 @@ private void checkIdleSlot() {
 					componentMainThreadExecutor,
 					(Acknowledge ignored, Throwable throwable) -> {
 						if (throwable != null) {
-							if (registeredTaskManagers.contains(expiredSlot.getTaskManagerId())) {
-								log.debug("Releasing slot [{}] of registered TaskExecutor {} failed. " +
-										"Trying to fulfill a different slot request.", allocationID, expiredSlot.getTaskManagerId(),
-									throwable);
-								tryFulfillSlotRequestOrMakeAvailable(expiredSlot);
-							} else {
-								log.debug("Releasing slot [{}] failed and owning TaskExecutor {} is no " +
-									"longer registered. Discarding slot.", allocationID, expiredSlot.getTaskManagerId());
-							}
+							// The slot status will be synced to task manager in next heartbeat.
+							log.debug("Releasing slot [{}] of registered TaskExecutor {} failed. Discarding slot.",
+										allocationID, expiredSlot.getTaskManagerId(), throwable);
```

Comment:
Shall we try to `freeSlot` for a fixed number of retries?

Reply:
I think it is not necessary now as the slot will soon be synced through heartbeat (in less than 10s).